lib/streamlit/runtime/pages_manager.py

```diff
@@ -15,322 +15,119 @@
-    ) -> None:
-        self._intended_page_script_hash = page_script_hash
-        self._intended_page_name = page_name
-
-    def get_initial_active_script(
-        self, page_script_hash: PageHash, page_name: PageName
-    ) -> PageInfo | None:
-        return self.pages_strategy.get_initial_active_script(
-            page_script_hash, page_name
-        )
-
-    def get_pages(self) -> dict[PageHash, PageInfo]:
-        return self.pages_strategy.get_pages()
-
-    def set_pages(self, pages: dict[PageHash, PageInfo]) -> None:
-        # Manually setting the pages indicates we are using MPA v2.
-        if isinstance(self.pages_strategy, PagesStrategyV1):
-            if os.path.exists(self.main_script_parent / "pages"):
-                _LOGGER.warning(
-                    "st.navigation was called in an app with a pages/ directory. "
-                    "This may cause unusual app behavior. You may want to rename the "
-                    "pages/ directory."
-                )
-            PagesManager.DefaultStrategy = PagesStrategyV2
-            self.pages_strategy = PagesStrategyV2(self)
-
-        self.pages_strategy.set_pages(pages)
-
-    def get_page_script(self, fallback_page_hash: PageHash = "") -> PageInfo | None:
-        # We assume the pages strategy is V2 cause this is used
-        # in the st.navigation call, but we just swallow the error
-        try:
-            return self.pages_strategy.get_page_script(fallback_page_hash)
-        except NotImplementedError:
-            return None
-
     def register_pages_changed_callback(
-        self,
-        callback: Callable[[str], None],
+        self, callback: Callable[[str], None]
     ) -> Callable[[], None]:
-        """Register a callback to be called when the set of pages changes.
-
-        The callback will be called with the path changed.
-        """
+        if self._is_mpa_v1:
+            return source_util.register_pages_changed_callback(callback)
 
-        return self.pages_strategy.register_pages_changed_callback(callback)
+        return lambda: None
 
-    def get_page_script_byte_code(self, script_path: str) -> Any:
-        if self._script_cache is None:
-            # Returning an empty string for an empty script
-            return ""
+    def find_page_info(
+        self,
+        page_script_hash: PageHash,
+        page_name: PageName | None,
+        *,
+        fallback_page_hash=None,
+    ) -> PageInfo | None:
+        # If a fallback hash is not provided, there's not much we can do, so we default to the main script hash
+        if fallback_page_hash is None:
+            fallback_page_hash = self.main_script_hash
+        # We want to check if the page_script_hash is set to some non-empty string
+        if page_script_hash:
+            return self.get_page_script_by_hash(page_script_hash)
+        # We allow page_name to have an empty string because that represents the main page
+        elif page_name is not None:
```

Comment:
nit: `elif` -> `if` there is no point in using `else` because we return in the if-branch if it is true. Not using `else` makes it more obvious that something happens in the `if` branch that can change the execution flow.